sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/OpenTelemetryAssertions.java

```diff
@@ -111,6 +117,107 @@ public static EventDataAssert assertThat(EventData eventData) {
+   * value satisfying {@code assertion}.
+   */
+  public static AttributeAssertion satisfies(
+      AttributeKey<String> key, StringAssertConsumer assertion) {
+    return AttributeAssertion.create(key, assertion);
+  }
+
+  /**
+   * Returns an {@link AttributeAssertion} that asserts the given {@code key} is present with a
+   * value satisfying {@code assertion}.
+   */
+  public static AttributeAssertion satisfies(
+      AttributeKey<Boolean> key, BooleanAssertConsumer assertion) {
+    return AttributeAssertion.create(key, assertion);
+  }
+
+  /**
+   * Returns an {@link AttributeAssertion} that asserts the given {@code key} is present with a
+   * value satisfying {@code assertion}.
+   */
+  public static AttributeAssertion satisfies(AttributeKey<Long> key, LongAssertConsumer assertion) {
+    return AttributeAssertion.create(key, assertion);
+  }
+
+  /**
+   * Returns an {@link AttributeAssertion} that asserts the given {@code key} is present with a
+   * value satisfying {@code assertion}.
+   */
+  public static AttributeAssertion satisfies(
+      AttributeKey<Double> key, DoubleAssertConsumer assertion) {
+    return AttributeAssertion.create(key, assertion);
+  }
+
+  /**
+   * Returns an {@link AttributeAssertion} that asserts the given {@code key} is present with a
+   * value satisfying {@code assertion}.
+   */
+  // Will require a cast only if using a custom implementation of AttributeKey which is highly
+  // unusual usage.
+  @SuppressWarnings("FunctionalInterfaceClash")
+  public static AttributeAssertion satisfies(
+      AttributeKey<List<String>> key, StringListAssertConsumer assertion) {
+    return AttributeAssertion.create(key, assertion);
+  }
+
+  /**
+   * Returns an {@link AttributeAssertion} that asserts the given {@code key} is present with a
+   * value satisfying {@code assertion}.
+   */
+  public static AttributeAssertion satisfies(
+      AttributeKey<List<Boolean>> key, BooleanListAssertConsumer assertion) {
+    return AttributeAssertion.create(key, assertion);
+  }
+
+  /**
+   * Returns an {@link AttributeAssertion} that asserts the given {@code key} is present with a
+   * value satisfying {@code assertion}.
+   */
+  public static AttributeAssertion satisfies(
+      AttributeKey<List<Long>> key, LongListAssertConsumer assertion) {
+    return AttributeAssertion.create(key, assertion);
+  }
+
+  /**
+   * Returns an {@link AttributeAssertion} that asserts the given {@code key} is present with a
+   * value satisfying {@code assertion}.
+   */
+  public static AttributeAssertion satisfies(
+      AttributeKey<List<Double>> key, DoubleListAssertConsumer assertion) {
+    return AttributeAssertion.create(key, assertion);
+  }
+
+  /**
+   * Returns an {@link AttributeAssertion} that asserts the given {@code key} is present with the
+   * given {@code value}.
+   */
+  public static <T> AttributeAssertion attribute(AttributeKey<T> key, T value) {
```

Comment:
I had initially named `equals` but conflicts with `Test.equals()`, this or `equalTo` came to mind and this seemed better

Reply:
I slightly prefer equalTo, but don't feel too strong

Reply:
Yeah it does seem to correspond better with satisfies, switched to equalTo